app/models/user.rb

```diff
@@ -16,42 +16,44 @@ class User < Common::RedisStore
   redis_ttl REDIS_CONFIG['user_store']['each_ttl']
   redis_key :uuid
 
-  # id.me attributes
   attribute :uuid
-  attribute :email
-  attribute :first_name
-  attribute :middle_name
-  attribute :last_name
-  attribute :gender
-  attribute :birth_date
-  attribute :zip
-  attribute :ssn
-  attribute :loa
-  # These attributes are fetched by SAML::User in the saml_response payload
-  attribute :multifactor   # used by F/E to decision on whether or not to prompt user to add MFA
-  attribute :authn_context # used by F/E to handle various identity related complexities pending refactor
-  # FIXME: if MVI were decorated on usr vs delegated to @mvi, then this might not have been necessary.
-  attribute :mhv_icn # only needed by B/E not serialized in user_serializer
-  attribute :mhv_uuid # this is the cannonical version of MHV Correlation ID, provided by MHV sign-in users
-
-  # vaafi attributes
-  attribute :last_signed_in, Common::UTCTime
-
-  # mhv_last_signed_in used to determine whether we need to notify MHV audit logging
-  # This is set to Time.now when any MHV session is first created, and nulled, when logout
-  attribute :mhv_last_signed_in, Common::UTCTime
+  attribute :last_signed_in, Common::UTCTime # vaafi attributes
+  attribute :mhv_last_signed_in, Common::UTCTime # MHV audit logging
 
   validates :uuid, presence: true
-  validates :email, presence: true
-  validates :loa, presence: true
 
-  # conditionally validate if user is LOA3
-  with_options(on: :loa3_user) do |user|
```

Comment:
Will there be data validation elsewhere?  If we don't get what we expect, it's best to catch it earlier than later.

Reply:
I think it will be up to the individual apps to have to figure that out longer term based on attributes that are available. Attributes can come from multiple different sources based on order of precedence. 

In fact, think about it this way. I could be a DS Logon user with `first_name: Kim, last_name: Jenkins` if this comes from ID.me that's what we currently treat as canonical. What if MVI has my old last name, from before I got married? When I sign-in via MHV, my identity will have `first_name: Kim, last_name: something_else` This is a problem I think. We've always said MVI is the source of truth, so why then is the saml response the source of truth? I think the SAML response ought to be the fallback when querying MVI fails (because user does not have adequate credentials to query it, MVI is down, etc). But I'd like to hear thoughts for against this.

As for validations though, I don't really think it makes sense to validate on these. It should really be up to the individual apps that depend on them to have handling for the event in which these values are null.

Reply:
The Facebook / Google analogy doesn't really work here in my opinion. It's true Facebook might have one last name and Google might have another, but with vets.gov we've said all along that MVI (or in the future vets360) is the source of truth, so the SAML response values really ought to only help us get to the source of truth when possible.

Reply:
> so the SAML response values really ought to only help us get to the source of truth when possible
This is more for future work (and came up un the MHV launch discussion)... but I think having MVI  (or 360 later) be the source of truth makes the most sense. ID.me is almost the edge case now. Querying by ICN (primary key) should be the first choice. Then by attributes if we don't have ICN. The provider is just the provider they can verify a user with some attributes is at a LOA level. The profile on hand at the VA, name on file, veteran status, military history etc. comes from another domain.